File: integrations/src/biomemakeover/java/juuxel/adorn/compat/biomemakeover/BiomeMakeoverCompat.java
package juuxel.adorn.compat.biomemakeover;

import com.google.auto.service.AutoService;
import juuxel.adorn.block.variant.BlockVariant;
import juuxel.adorn.block.variant.BlockVariantSet;
import juuxel.adorn.block.variant.CompatBlockVariantSet;

import java.util.List;

@AutoService(BlockVariantSet.class)
public final class BiomeMakeoverCompat extends CompatBlockVariantSet {
    @Override
    protected String getModId() {
        return "biomemakeover";
    }

    @Override
    public List<BlockVariant> getWoodVariants() {
        return createVariants(
            BlockVariant.Wood::new,
            "ancient_oak",
            "blighted_balsa",
            "willow",
            "swamp_cypress"
        );
    }
}

File: forge/src/main/kotlin/juuxel/adorn/platform/forge/compat/BiomeMakeoverCompat.kt
package juuxel.adorn.platform.forge.compat

import juuxel.adorn.block.variant.BlockVariant
import juuxel.adorn.block.variant.BlockVariantSet

object BiomeMakeoverCompat : BlockVariantSet {
    override val woodVariants = listOf(
        "ancient_oak",
        "blighted_balsa",
        "willow",
        "swamp_cypress",
    ).map { BlockVariant.Wood("biomemakeover/$it") }
}
